Empire/Empire.cpp

```diff
@@ -1036,147 +1110,70 @@ void ProductionQueue::Update() {
-    dp_time_start = boost::posix_time::ptime(boost::posix_time::microsec_clock::local_time()); 
-
-    //invert lookup direction of sim_queue_element_groups:
-    std::map< std::set<int>, std::vector<int>  > elements_by_group;
-    for (unsigned int i = 0; i < sim_queue.size(); ++i)
-        elements_by_group[queue_element_groups[i]].push_back(i);
-
-    // within each group, allocate PP to queue items
-    for (const std::map<std::set<int>, float>::value_type& group : available_pp) {
-        unsigned int first_turn_pp_available = 1; //the first turn any pp in this resource group is available to the next item for this group
-        unsigned int turn_jump = 0;
-        //pp_still_available[turn-1] gives the PP still available in this resource pool at turn "turn"
-        std::vector<float> pp_still_available(TOO_MANY_TURNS, group.second);  // initialize to the groups full PP allocation for each turn modeled
-
-        std::vector<int> &this_group_elements = elements_by_group[group.first];
-        std::vector<int>::const_iterator group_begin = this_group_elements.begin();
-        std::vector<int>::const_iterator group_end = this_group_elements.end();
-
-        // cycle through items on queue, if in this resource group then allocate production costs over time against those available to group
-        for (std::vector<int>::const_iterator el_it = group_begin;
-             (el_it != group_end) && ((boost::posix_time::ptime(boost::posix_time::microsec_clock::local_time())-dp_time_start).total_microseconds()*1e-6 < TOO_LONG_TIME);
-             ++el_it)
-        {
-            first_turn_pp_available += turn_jump;
-            turn_jump = 0;
-            if (first_turn_pp_available > TOO_MANY_TURNS) {
-                DebugLogger()  << "ProductionQueue::Update: Projections for Resource Group halted at " 
-                               << TOO_MANY_TURNS << " turns; remaining items in this RG marked completing 'Never'.";
-                break; // this resource group is allocated-out for span of simulation; remaining items in group left as never completing
-            }
-
-            unsigned int i = *el_it;
-            ProductionQueue::Element& element = sim_queue[i];
-            if (element.paused)
+    boost::posix_time::ptime sim_time_start;
+    boost::posix_time::ptime sim_time_end;
+    long sim_time;
+    sim_time_start = boost::posix_time::ptime(boost::posix_time::microsec_clock::local_time()); 
+    std::map<std::set<int>, float>  allocated_pp;
+    float sim_available_stockpile = available_stockpile;
+    std::map<std::set<int>, float>  allocated_stockpile_pp;
+    int dummy_int = 0;
+    
+    for (int sim_turn = 1; sim_turn <= TOO_MANY_TURNS; sim_turn ++) {
+        if ((boost::posix_time::ptime(boost::posix_time::microsec_clock::local_time()) -
+                    sim_time_start).total_microseconds()*1e-6 >= TOO_LONG_TIME)
+            break;
```

Comment:
after a multi-line if, particularly where the later lines are indented from the first, put `{` and `}` around the following indented code